### DISCUSSION TITLE
Change ${CMAKE_SOURCE_DIR} to ${CMAKE_CURRENT_SOURCE_DIR}

### DISCUSSION THREAD

**Discussion by Magitrop** [gabime/spdlog#2954] [General]
CMakeLists.txt, lines 294-298:
```
if(CMAKE_GENERATOR MATCHES "Visual Studio")
    source_group(TREE ${CMAKE_SOURCE_DIR}/include PREFIX include FILES ${SPDLOG_HEADERS})
    source_group(TREE ${CMAKE_SOURCE_DIR}/src PREFIX sources FILES ${SPDLOG_SRCS})    
    source_group(sources FILES ${VERSION_RC})
endif()
```
The structure of my project was:
```
home
|--- dependencies
     |--- spdlog
     |--- ...
|--- code
     |--- ...
```
CMake tried but was not able to find `home/include` as a prefix.
In order to fix it I had to change `${CMAKE_SOURCE_DIR}` to `${CMAKE_CURRENT_SOURCE_DIR}`.
It seems to me that this fix should be done to master.

**Comment by tt4g** [comment]
There is no such code in the current main branch v1.x.

I assume you are referring to the code in `CMakeLists.txt` on v2.x branch.
v2.x branch is not stable, so could you please include the branch name in the subject?

**Reply by tt4g** [reply]
PS
PR is welcome.
